src/merge.rs

```diff
@@ -959,13 +1131,32 @@ impl<'t, D: Driver, A: AbortSignal> Merger<'t, D, A> {
                         Validity::Replace => ConflictResolution::Local,
                     }
                 };
+                let children = if local_node.has_matching_children(remote_node) {
+                    ConflictResolution::Unchanged
+                } else {
+                    ConflictResolution::Remote
+                };
                 // For children, we always use the remote side.
-                (item, ConflictResolution::Remote)
+                (item, children)
             }
 
             (false, false) => {
-                // The item is unchanged on both sides.
-                (ConflictResolution::Unchanged, ConflictResolution::Unchanged)
+                // The item's children differ, even though it's not flagged as
+                // unmerged, so we prefer the newer side for children.
```

Comment:
This comment is in the wrong place, it should be in the `else if` where we set `children` below.